src/graphql/root/mutation/user-logout.ts

```diff
@@ -0,0 +1,40 @@
+import { GT } from "@graphql/index"
+
+import AuthToken from "@graphql/types/scalar/auth-token"
+import AuthTokenPayload from "@graphql/types/payload/auth-token"
+import { AuthWithPhonePasswordlessService } from "@services/kratos"
+import { mapAndParseErrorForGqlResponse } from "@graphql/error-map"
+
+const UserLogoutInput = GT.Input({
+  name: "UserLogoutInput",
+  fields: () => ({
+    authToken: {
+      type: GT.NonNull(AuthToken),
+    },
+  }),
+})
+
+const UserLogoutMutation = GT.Field<{
+  input: {
+    authToken: string | InputValidationError
+  }
+}>({
+  extensions: {
+    complexity: 120,
+  },
+  type: GT.NonNull(AuthTokenPayload),
+  args: {
+    input: { type: GT.NonNull(UserLogoutInput) },
+  },
+  resolve: async (_, args) => {
+    const { authToken } = args.input
+    if (authToken instanceof Error) return
+    const authService = AuthWithPhonePasswordlessService()
+    const logoutResp = await authService.logout(authToken)
+    if (logoutResp instanceof Error)
```

Comment:
You need to create a use case for it in apps. Graphql code should not call service code directly